src/proto/h1/decode.rs

```diff
@@ -143,45 +150,72 @@ impl Decoder {
                     } else {
                         *remaining -= num;
                     }
-                    Poll::Ready(Ok(buf))
+                    Poll::Ready(Ok(Frame::data(buf)))
                 }
             }
             Chunked {
                 ref mut state,
                 ref mut chunk_len,
                 ref mut extensions_cnt,
+                ref mut trailers_buf,
+                ref mut trailers_cnt,
             } => {
                 loop {
                     let mut buf = None;
                     // advances the chunked state
-                    *state = ready!(state.step(cx, body, chunk_len, extensions_cnt, &mut buf))?;
+                    *state = ready!(state.step(
+                        cx,
+                        body,
+                        chunk_len,
+                        extensions_cnt,
+                        &mut buf,
+                        trailers_buf,
+                        trailers_cnt,
+                    ))?;
                     if *state == ChunkedState::End {
                         trace!("end of chunked");
-                        return Poll::Ready(Ok(Bytes::new()));
+
+                        if trailers_buf.is_some() {
+                            trace!("found possible trailers");
+                            match decode_trailers(
+                                &mut trailers_buf.take().expect("Trailer is None"),
+                                // decoder enforces that trailers count will not exceed usize::MAX
```

Comment:
where are we enforcing that?

Reply:
here: https://github.com/hyperium/hyper/pull/3637/files#diff-2ef0d3180cacbd6f072b1728bd8c8a9d0c132df381876f264b67e18bb4a87d5fR506

Reply:
i also added a check in [545f8c4](https://github.com/hyperium/hyper/pull/3637/commits/545f8c48d3da589cf7a4154aa6433db1bbe07076)